From 922c3b19cfbfeaa8e4ffc7a1ff8779a2a91a6c19 Mon Sep 17 00:00:00 2001
From: Rob Browning <rlb@defaultvalue.org>
Date: Sun, 18 Mar 2012 13:28:24 -0500
Subject: [PATCH] Mark "mutex with owner not retained" threads test as
 unresolved.

Upstream-Status: Inappropriate [debian patch]

Signed-off-by: Nitin A Kamble <nitin.a.kamble@intel.com>

As per discussion with upstream, mark this test as unresolved since it
may produce false negatives, depending on the behavior/timing of the
garbage collector.

---
 test-suite/tests/threads.test | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/test-suite/tests/threads.test b/test-suite/tests/threads.test
index 3b7a3e4..c6fe81e 100644
--- a/test-suite/tests/threads.test
+++ b/test-suite/tests/threads.test
@@ -423,8 +423,10 @@

             (gc) (gc)
             (let ((m (g)))
-              (and (mutex? m)
-                   (eq? (mutex-owner m) (current-thread)))))))
+              (or
+               (and (mutex? m)
+                    (eq? (mutex-owner m) (current-thread)))
+               (throw 'unresolved))))))

       ;;
       ;; mutex lock levels
